src/routes/preHandlers/subPrograms.js

```diff
@@ -25,13 +25,15 @@ exports.authorizeStepAdd = async (req) => {
 exports.authorizeSubProgramUpdate = async (req) => {
   const subProgram = await SubProgram.findOne({ _id: req.params._id })
     .populate({ path: 'program', select: '_id' })
-    .populate({ path: 'steps', select: '_id type' })
+    .populate({ path: 'steps', select: '_id type', populate: { path: 'activities', populate: 'cards' } })
```

Comment:
Si je populate activities avec type et card avec template, mes virtuals deviennent faux (isCardValid devient false alors qu'il est censé être true et qu'il est bien true avec ce populate).
Je n'arrive pas à comprendre pourquoi :/

Reply:
c'est normal :  tu as besoin de tous les champs des cartes pour savoir si elles sont valides ou non ? pourquoi ne veux tu que le `template`
et le `type` ne sert a rien pour la validité des activités... donc dans model `Step` on ferait meiux de verifier si `cards` est bien dans l'objet pour calculer `areCardsValid`

Reply:
Donc : `this.steps && this.steps.length && this.steps[0].activities && this.steps[0].activities[0].cards` ?

Reply:
il faut le changer dans Steps aussi

Reply:
up sur ce commentaire